src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -2090,23 +2115,39 @@ private MethodData FindBinaryOperator(string operatorName, Expression left, Expr
 
 		private static Expression GenerateStringConcat(Expression left, Expression right)
 		{
-			var concatMethod = typeof(string).GetMethod("Concat", new[] { typeof(object), typeof(object) });
-			if (concatMethod == null)
-				throw new Exception("String concat method not found");
-
-			var rightObj =
-				right.Type.IsValueType
-				? Expression.ConvertChecked(right, typeof(object))
-				: right;
-			var leftObj =
-				left.Type.IsValueType
-				? Expression.ConvertChecked(left, typeof(object))
-				: left;
+			var leftObj = GenerateStringConcatOperand(left);
+			var rightObj = GenerateStringConcatOperand(right);
 
 			return Expression.Call(
-					null,
-					concatMethod,
-					new[] { leftObj, rightObj });
+				null,
+				_concatMethod,
+				new[] { leftObj, rightObj });
+		}
+
+		private static Expression ToStringOrNull(Expression expression)
+		{
+			var nullConstant = Expression.Constant(null);
+			Expression condition = Expression.Equal(
+				expression,
+				nullConstant);
+
+			return Expression.Condition(
+				condition,
+				Expression.ConvertChecked(nullConstant, typeof(String)),
```

Comment:
Maybe here you can use directly `nullConstant`? without any conversion? At the end `null` is a valid string....

Reply:
I mean writing directly:
```
return Expression.Condition(
				condition,
				nullConstant,
				Expression.Call(expression, _toStringMethod));
```

It should be the equivalent of:
```
if (value == null) {
  return null;
}
else {
  return value.ToString();
}
```

Reply:
Unfortunately it is not possible 😞 I tried `nullConstant` without Type definition but `Expression.Condition` required that `then` and `else` if-branches should be of the same type.